### PR TITLE
 python39Packages.python-openstackclient: init at 5.6.0 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
replaces #134282
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/openstacksdk/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    requests-mock
+    stestr
+    testscenarios
+  ];
+
+  checkPhase = ''
+    stestr run -e <(echo "
+    openstack.tests.unit.cloud.test_image.TestImage.test_create_image_task
+    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_error_396
+    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_wait
```
Comment (angustrau) [inline pkgs/development/python-modules/openstacksdk/default.nix:65]: This test was also failing on my machine

```suggestion
    openstack.tests.unit.image.v2.test_proxy.TestImageProxy.test_wait_for_task_wait
    openstack.tests.unit.test_resource.TestWaitForStatus.test_status_fails
```

File: pkgs/development/python-modules/osc-lib/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ianpk32vwdllpbk4zhfifqb5b064cbmia2hm02lcrh2m76z0zi5";
+  };
+
+  nativeBuildInputs = [
+    pbr
+  ];
+
+  propagatedBuildInputs = [
```
Comment (angustrau) [inline pkgs/development/python-modules/osc-lib/default.nix:26]:
```suggestion
  propagatedBuildInputs = [
      cliff
```

**Review by emilytrau — CHANGES_REQUESTED** [review]
Thanks so much for cleaning this up! I've tested this on my machine. Would `openstackclient` and maybe `stestr` belong in `all-packages.nix`?

File: pkgs/applications/misc/zk-shell/default.nix [hunk 3]
```diff
@@ -1,8 +1,8 @@
-{ lib, fetchFromGitHub, pythonPackages }:
+{ lib, fetchFromGitHub, python3Packages }:
 
-pythonPackages.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
+  pname = "zk-shell";
```
Comment (superherointj) [inline pkgs/applications/misc/zk-shell/default.nix:4]: This `zk-shell` package seems to be added here by mistake.
Comes from PR #137350 from SuperSandro2000/zk-shell

File: pkgs/development/python-modules/debtcollector/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "c7a9fac814ab5904e23905516b18356cc907e7d27c05da58d37103f001967846";
+  };
+
+  nativeBuildInputs = [ pbr ];
+
+  propagatedBuildInputs = [ six wrapt ];
+
+  doCheck = false;
```
Comment (risicle) [inline pkgs/development/python-modules/debtcollector/default.nix:16]: Would be nice to leave a small comment explaining the need for the recursion-free test-running technique when we do this.

**Comment by risicle** [comment]
Numerous test failures for eventlet on macos 10.15, starting with:

```
running install tests
.............S..EFFTraceback (most recent call last):
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/hubs/kqueue.py", line 105, in wait
    readers.get(fileno, hub.noop).cb(fileno)
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/convenience.py", line 135, in serve
    conn, addr = sock.accept()
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/greenio/base.py", line 230, in accept
    self._trampoline(fd, read=True, timeout=self.gettimeout(), timeout_exc=_timeout_exc)
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/greenio/base.py", line 208, in _trampoline
    return trampoline(fd, read=read, write=write, timeout=timeout,
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/hubs/__init__.py", line 159, in trampoline
    return hub.switch()
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/hubs/hub.py", line 313, in switch
    return self.greenlet.switch()
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/convenience.py", line 91, in _stop_checker
    t.wait()
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/greenthread.py", line 181, in wait
    return self._exit_event.wait()
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/event.py", line 132, in wait
    current.throw(*self._exc)
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/eventlet/greenthread.py", line 221, in main
    result = function(*args, **kwargs)
  File "/private/tmp/nix-build-python3.8-eventlet-0.32.0.drv-0/eventlet-0.32.0/tests/convenience_test.py", line 41, in crasher
    0 // 0
ZeroDivisionError: integer division or modulo by zero
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 137368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.zake</li>
<li>python39Packages.zake</li>
</ul>
</details>
<details>
<summary>
62 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>home-assistant</li>
<li>onionshare</li>
<li>onionshare-gui</li>
<li>openstackclient (python39Packages.python-openstackclient)</li>
<li>python38Packages.aioambient</li>
<li>python38Packages.cassandra-driver</li>
<li>python38Packages.debtcollector</li>
<li>python38Packages.eventlet</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.hacking</li>
<li>python38Packages.kazoo</li>
<li>python38Packages.keystoneauth1</li>
<li>python38Packages.openstacksdk</li>
<li>python38Packages.os-service-types</li>
<li>python38Packages.osc-lib</li>
<li>python38Packages.oslo-concurrency</li>
<li>python38Packages.oslo-config</li>
<li>python38Packages.oslo-context</li>
<li>python38Packages.oslo-db</li>
<li>python38Packages.oslo-i18n</li>
<li>python38Packages.oslo-log</li>
<li>python38Packages.oslo-serialization</li>
<li>python38Packages.oslo-utils</li>
<li>python38Packages.oslotest</li>
<li>python38Packages.python-cinderclient</li>
<li>python38Packages.python-engineio</li>
<li>python38Packages.python-keystoneclient</li>
<li>python38Packages.python-novaclient</li>
<li>python38Packages.python-openstackclient</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.stestr</li>
<li>python38Packages.subunit2sql</li>
<li>python39Packages.aioambient</li>
<li>python39Packages.cassandra-driver</li>
<li>python39Packages.debtcollector</li>
<li>python39Packages.eventlet</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.hacking</li>
<li>python39Packages.kazoo</li>
<li>python39Packages.keystoneauth1</li>
<li>python39Packages.openstacksdk</li>
<li>python39Packages.os-service-types</li>
<li>python39Packages.osc-lib</li>
<li>python39Packages.oslo-concurrency</li>
<li>python39Packages.oslo-config</li>
<li>python39Packages.oslo-context</li>
<li>python39Packages.oslo-db</li>
<li>python39Packages.oslo-i18n</li>
<li>python39Packages.oslo-log</li>
<li>python39Packages.oslo-serialization</li>
<li>python39Packages.oslo-utils</li>
<li>python39Packages.oslotest</li>
<li>python39Packages.python-cinderclient</li>
<li>python39Packages.python-engineio</li>
<li>python39Packages.python-keystoneclient</li>
<li>python39Packages.python-novaclient</li>
<li>python39Packages.python-socketio</li>
<li>python39Packages.stestr</li>
<li>python39Packages.subunit2sql</li>
<li>zk-shell</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 137368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.zake</li>
<li>python39Packages.zake</li>
</ul>
</details>
<details>
<summary>
62 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>home-assistant</li>
<li>onionshare</li>
<li>onionshare-gui</li>
<li>openstackclient (python39Packages.python-openstackclient)</li>
<li>python38Packages.aioambient</li>
<li>python38Packages.cassandra-driver</li>
<li>python38Packages.debtcollector</li>
<li>python38Packages.eventlet</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.hacking</li>
<li>python38Packages.kazoo</li>
<li>python38Packages.keystoneauth1</li>
<li>python38Packages.openstacksdk</li>
<li>python38Packages.os-service-types</li>
<li>python38Packages.osc-lib</li>
<li>python38Packages.oslo-concurrency</li>
<li>python38Packages.oslo-config</li>
<li>python38Packages.oslo-context</li>
<li>python38Packages.oslo-db</li>
<li>python38Packages.oslo-i18n</li>
<li>python38Packages.oslo-log</li>
<li>python38Packages.oslo-serialization</li>
<li>python38Packages.oslo-utils</li>
<li>python38Packages.oslotest</li>
<li>python38Packages.python-cinderclient</li>
<li>python38Packages.python-engineio</li>
<li>python38Packages.python-keystoneclient</li>
<li>python38Packages.python-novaclient</li>
<li>python38Packages.python-openstackclient</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.stestr</li>
<li>python38Packages.subunit2sql</li>
<li>python39Packages.aioambient</li>
<li>python39Packages.cassandra-driver</li>
<li>python39Packages.debtcollector</li>
<li>python39Packages.eventlet</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.hacking</li>
<li>python39Packages.kazoo</li>
<li>python39Packages.keystoneauth1</li>
<li>python39Packages.openstacksdk</li>
<li>python39Packages.os-service-types</li>
<li>python39Packages.osc-lib</li>
<li>python39Packages.oslo-concurrency</li>
<li>python39Packages.oslo-config</li>
<li>python39Packages.oslo-context</li>
<li>python39Packages.oslo-db</li>
<li>python39Packages.oslo-i18n</li>
<li>python39Packages.oslo-log</li>
<li>python39Packages.oslo-serialization</li>
<li>python39Packages.oslo-utils</li>
<li>python39Packages.oslotest</li>
<li>python39Packages.python-cinderclient</li>
<li>python39Packages.python-engineio</li>
<li>python39Packages.python-keystoneclient</li>
<li>python39Packages.python-novaclient</li>
<li>python39Packages.python-socketio</li>
<li>python39Packages.stestr</li>
<li>python39Packages.subunit2sql</li>
<li>zk-shell</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
If I missed something and something does not work please send PRs. I'd rather merge this in almost working state rather than keep it open for longer.

@superherointj now you can add further packages. Send me a dm if you need help with something or there are questions.

**Pull request merged**